spring-vault-core/src/main/java/org/springframework/vault/authentication/JwtAuthenticationOptions.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2017-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.vault.authentication;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.springframework.util.Assert;
+
+/**
+ * Authentication options for {@link JwtAuthentication}.
+ * <p>
+ * Authentication options provide the role and the JWT. {@link JwtAuthenticationOptions}
+ * can be constructed using {@link #builder()}. Instances of this class are immutable once
+ * constructed.
+ * <p>
+ *
+ * @author Nanne Baars
+ * @since 3.0.4
+ * @see JwtAuthentication
+ * @see #builder()
+ */
+public class JwtAuthenticationOptions {
+
+	/**
+	 * Path of the JWT authentication backend mount. Optional and defaults to
+	 * {@literal jwt}.
+	 */
+	private final Optional<String> path;
+
+	/**
+	 * Name of the role against which the login is being attempted. Defaults to configured
+	 * default_role if not provided. See
+	 * <a href="https://developer.hashicorp.com/vault/api-docs/auth/jwt#configure">Vault
+	 * JWT configuration</a>
+	 */
+	private final Optional<String> role;
+
+	/**
+	 * The signed JSON Web Token (JWT) to be used for authentication.
+	 */
+	private final String jwt;
```

Comment:
This should be a supplier to allow integrating with JWT's obtained from somewhere else, which can expire over time.

Reply:
Ah I had this in the first implementation I will update this

Reply:
Updated and moved to supplier